libpod/container_internal_linux_test.go

```diff
@@ -97,3 +102,54 @@ func TestAppendLocalhost(t *testing.T) {
 		assert.Equal(t, "127.0.0.1\tlocalhost", c.appendLocalhost("127.0.0.1\tlocalhost"))
 	}
 }
+
+func TestContainerUUIDEnv(t *testing.T) {
+	ctx := context.Background()
+	c := Container{
+		config: &ContainerConfig{
+			Spec: &spec.Spec{
+				Process: &spec.Process{},
+				Root:    &spec.Root{},
+				Linux: &spec.Linux{
+					Namespaces: []spec.LinuxNamespace{{Type: spec.NetworkNamespace, Path: ""}},
+				},
+			},
+			ID: stringid.GenerateNonCryptoID(),
+			IDMappings: storage.IDMappingOptions{
+				UIDMap: []idtools.IDMap{{ContainerID: 0, HostID: os.Geteuid(), Size: 1}},
+				GIDMap: []idtools.IDMap{{ContainerID: 0, HostID: os.Getegid(), Size: 1}},
+			},
+			ContainerNetworkConfig: ContainerNetworkConfig{UseImageHosts: true},
+		},
+		state: &ContainerState{
+			BindMounts: map[string]string{"/run/.containerenv": ""},
+			RunDir:     t.TempDir(),
+		},
+		runtime: &Runtime{
+			config: &config.Config{Containers: config.ContainersConfig{}},
+		},
+	}
+
+	spec, err := c.generateSpec(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.False(t, hasContainerUUIDEnv(t, &c, spec))
+
+	c.config.Systemd = true
+	spec, err = c.generateSpec(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.True(t, hasContainerUUIDEnv(t, &c, spec))
+}
+
+func hasContainerUUIDEnv(t *testing.T, c *Container, spec *spec.Spec) bool {
+	t.Helper()
+	for _, checkEnv := range spec.Process.Env {
+		if checkEnv == "container_uuid="+c.Config().ID[:32] {
```

Comment:
```suggestion
		if checkEnv == "container_uuid="+c.config.ID[:32] {
```
This should speed up testing